engine/swarm/secrets.md

```diff
@@ -0,0 +1,837 @@
+
+    ```bash
+    $ docker service create \
+         --name wordpress \
+         --replicas 1 \
+         --network mysql_private \
+         --publish 30000:80 \
+         --mount type=volume,source=wpdata,destination=/var/www/html \
+         --secret source=mysql_password,target=wp_db_password,mode=0400 \
+         -e WORDPRESS_DB_USER="wordpress" \
+         -e WORDPRESS_DB_PASSWORD_FILE="/run/secrets/wp_db_password" \
+         -e WORDPRESS_DB_HOST="mysql:3306" \
+         -e WORDPRESS_DB_NAME="wordpress" \
+         wordpress:latest
+    ```
+
+6.  Verify the service is running using `docker service ls` and
+    `docker service ps` commands.
+
+    ```bash
+    $ docker service ls
+
+    ID            NAME       MODE        REPLICAS  IMAGE
+    wvnh0siktqr3  mysql      replicated  1/1       mysql:latest
+    nzt5xzae4n62  wordpress  replicated  1/1       wordpress:latest
+    ```
+
+    ```bash
+    $ docker service ps wordpress
+
+    ID            NAME         IMAGE             NODE  DESIRED STATE  CURRENT STATE           ERROR  PORTS
+    aukx6hgs9gwc  wordpress.1  wordpress:latest  moby  Running        Running 52 seconds ago   
+    ```
+
+    At this point, you could actually revoke the WordPress service's access to
+    the `mysql_password` secret, because WordPress has copied the secret to its
+    configuration file `wp-config.php`. Don't do that for now, because we will
+    use it later to facilitate rotating the MySQL password.
+
+7.  Access `http://localhost:30000/` from any swarm node and set up WordPress
+    using the web-based wizard. All of these settings are stored in the MySQL
+    `wordpress` database. WordPress automatically generates a password for your
+    WordPress user, which is completely different from the password WordPress
+    uses to access MySQL. Store this password securely, such as in a password
+    manager. You will need it to log into WordPress after
+    [rotating the secret](#example-rotate-a-secret).
+
+    Go ahead and write a blog post or two and install a WordPress plugin or
+    theme to verify that WordPress is fully operational and its state is saved
+    across service restarts.
+
+8.  Do not clean up any services or secrets if you intend to proceed to the next
+    example, which demonstrates how to rotate the MySQL root password.
+
+
+### Example: Rotate a secret
+
+This example builds upon the previous one. In this scenario, you create a new
+secret with a new MySQL password, update the `mysql` and `wordpress` services to
+use it, then remove the old secret.
+
+**Note**: Changing the password on a MySQL database involves running extra
+queries or commands, as opposed to just changing a single environment variable
+or a file, since the image only sets the MySQL password if the database doesn’t
+already exist, and MySQL stores the password within a MySQL database by default.
+Rotating passwords or other secrets will often involve additional steps outside
```

Comment:
Nonblocking nitpick: should this be "may"?  If it's something simple like an API key the service just access, just changing the service spec should be fine.  Alternately, this is likely always the case with databases, so maybe we can call out DBs specifically here?